src/dynamic_set.rs

```diff
@@ -22,6 +24,20 @@ pub(crate) struct Set {
     buckets: Box<[Mutex<Option<Box<Entry>>>]>,
 }
 
+impl Set {
+    #[cfg(feature = "malloc_size_of")]
+    pub(crate) fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        self.buckets.shallow_size_of(ops) +
+            self.buckets
+                .iter()
+                .map(|bucket| {
+                    bucket.lock().as_ref().map_or(0, |entry| entry.size_of(ops))
+                })
+                .sum::<usize>()
+    }
+}
+
+#[cfg_attr(feature = "malloc_size_of", derive(MallocSizeOf))]
```

Comment:
I would personally be tempted to do a manual impl here to avoid the dep on the derive crate (but I suspect my opinion will be a minority one here so this is more of a "noted dissenting opinion" rather than a request for a change).